CenterForOpenScience/osf.io
# [#OSF-8781]Citations with suffixes show up incorrectly in preprints

## Purpose
Fix the problem that citations with suffixes show up incorrectly in preprints


## Changes
after change:
<img width="1116" alt="screen shot 2017-12-01 at 3 48 37 pm" src="https://user-images.githubusercontent.com/4974056/33502699-292053da-d6af-11e7-8273-9639034e067f.png">
<img width="1142" alt="screen shot 2017-12-01 at 3 48 43 pm" src="https://user-images.githubusercontent.com/4974056/33502700-292e0c00-d6af-11e7-8ef3-78d377e06900.png">
<img width="1129" alt="screen shot 2017-12-01 at 3 48 49 pm" src="https://user-images.githubusercontent.com/4974056/33502701-29370dc8-d6af-11e7-9e75-9230c2cb2386.png">


QA notes:
The short version for this is that test
staging-api.osf.io/v2/nodes/<node._id>/citation/apa/
staging-api.osf.io/v2/nodes/<node._id>/citation/chicago-author-date/
staging-api.osf.io/v2/nodes/<node._id>/citation/modern-language-association/

the citation return from the api should match the citation in project overview page citation widget.

Special shortcut for special change to notice for each citation format:
apa:
when 1 < contributor < 8,
contributor 1, contributor 2, contributor 3, & contributor 4 
when contributor >= 8,
contributor 1, contributor 2, contributor 3, contributor 4, contributor 5, contributor 6, ... contributor last one

chicago:
when 1 < contributor < 11,
contributor 1, contributor 2, contributor 3, and contributor 4
when contributor >= 11,
contributor 1, contributor 2, contributor 3, contributor 4, contributor 5, contributor 6, contributor 7, contributor 8, contributor9, contributor 10, et al.

mla:
when 1< contributor < 5,
contributor 1, contributor 2, and contributor 3
when contributor >=5m
contributor 1 et al.

## Side effects




## Ticket
https://openscience.atlassian.net/browse/OSF-8781
